### PR TITLE
scrapper: add functional scrapper

### PR DESCRIPTION
**Pull request by MrColorado**

### CODE REVIEW

File: src/scripts/anti-crise-scrapper.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import datetime
+from bs4 import BeautifulSoup
+from datetime import timedelta
+from src.models.Promotion import Promotion
+from src.scripts.utils import simple_get, promotion_post
+from src.scripts.scrapper import Scrapper
+
+USER_ID = 1
+HISTORY_FILE = "../misc/history"
```
Comment (hugodelahousse) [inline src/scripts/anti-crise-scrapper.py]: Relative paths are not great. You should can import Path from `pathlib` and do something like `Path(__file__).parent / 'misc/history'`

File: src/scripts/anti-crise-scrapper.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import datetime
+from bs4 import BeautifulSoup
+from datetime import timedelta
+from src.models.Promotion import Promotion
+from src.scripts.utils import simple_get, promotion_post
+from src.scripts.scrapper import Scrapper
+
+USER_ID = 1
```
Comment (hugodelahousse) [inline src/scripts/anti-crise-scrapper.py]: What's this ?

File: src/scripts/anti-crise-scrapper.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+HISTORY_FILE = "../misc/history"
+API_URL = "https://yakosa.herokuapp.com/"
```
Comment (hugodelahousse) [inline src/scripts/anti-crise-scrapper.py]: This should not be in this file

File: src/scripts/anti-crise-scrapper.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+URL = "http://anti-crise.fr/les-catalogues-avec-optimisations/"
```
Comment (hugodelahousse) [inline src/scripts/anti-crise-scrapper.py]: This should probably be an attribute of `AntiCriseScrapper`

File: src/scripts/anti-crise-scrapper.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+
+def get_history():
```
Comment (hugodelahousse) [inline src/scripts/anti-crise-scrapper.py]: Maybe move those functions to private methods of the scrapper

**Review by hugodelahousse — CHANGES_REQUESTED** [review]

File: src/scripts/anti-crise-scrapper.py (hunk 2 above)
Comment (MrColorado) [reply src/scripts/anti-crise-scrapper.py]: When we are creating a promotion it must be related to a user. I could simply put this foreign key as optional in the db.

File: src/scripts/scrapper.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    def transform(self, soup):
+        pass
+
+    @abstractmethod
+    def frequency(self, time):
+        pass
+
+    @abstractmethod
+    def run(self):
+        pass
```
Comment (hugodelahousse) [inline src/scripts/scrapper.py]:
```suggestion
        pass

```

**Review by hugodelahousse — CHANGES_REQUESTED** [review]

**Pull request merged**